### DISCUSSION TITLE
C# for markup in Blazor

### DISCUSSION THREAD

**Discussion by akhanalcs** [VincentH-Net/CSharpForMarkup#36] [General]
First of all, thank you so much for this amazing library you've put together!

I'd like to make a request:
Can you please bring this to Blazor?
We .NET web developers would rejoice to see this come to Blazor.

**Comment by VincentH-Net** [comment]
Thanks @affableashish! I am very interested to learn how many (potential) Blazor devs would want this. Do you have any indication?

My approach would be to generate a C# Markup api for HTML5 based on https://github.com/tawesoft/html5spec/blob/master/spec-json/elements.json , and then integrate into Blazor so you no longer have to mix HTML and C# in the same page - it can all be C#, and you can maintain separation between UI markup (declarative) and UI logic (imperative) in partial class files if you like.

I have not looked into how you can emit HTML from C# in Blazor - if you can point me in the right direction I would appreciate.

**Reply by akhanalcs** [reply]
Thank you Vincent! I apologize I don't really know how many potential Blazor Devs would want this because at this time not a lot of Web developers know that this is possible. After it becomes a thing and after you present this in few dotNET community standups, I'm sure a lot more developers would use this (I definitely would!).

I also do not know how to emit HTML from C# in Blazor, if someone in the community who has already tried something similar could answer that, that'd be really nice.
Also if you can talk to Microsoft folks (For eg: Daniel Roth) about it, they'd probably be able to help.
Thank you again for creating this amazing library!
